### PR TITLE
bazel: 5.2.0 -> 5.3.1

### PR DESCRIPTION
**Pull request by dmivankov**
###### Description of changes

https://blog.bazel.build/2022/08/23/bazel-5.3.html

to superceed https://github.com/NixOS/nixpkgs/pull/187943 (5.3.0)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/tensorflow/default.nix [hunk 1]
```diff
@@ -369,13 +369,13 @@ let
     fetchAttrs = {
       # cudaSupport causes fetch of ncclArchive, resulting in different hashes
       sha256 = if cudaSupport then
-        "sha256-KtVReqHL3zxE8TPrqIerSOt59Mgke/ftoFZKMzgX/u8="
+        "sha256-KtVReqHL3zxE8TPrqIerSOt59Mgke/ftoFZKMzgX/u1="
       else
         if stdenv.isDarwin then
           # FIXME: this checksum is currently wrong, since the tensorflow dependency fetch is broken on darwin
-          "sha256-j2k9Q+k41nq5nP1VjjkkNjXRov1uAda4RCMDMAthjr0="
+          "sha256-j2k9Q+k41nq5nP1VjjkkNjXRov1uAda4RCMDMAthjr1="
```
Comment (dmivankov) [inline pkgs/development/python-modules/tensorflow/default.nix:376]: this and one above is fake, will need to update

**Comment by dmivankov** [comment]
darwin tensorflow fails on something else about libtool
```
Auto-Configuration Error: Cannot find libtool or LIBTOOL; either correct your path or set the LIBTOOL environment variable
Analyzing: 2 targets (256 packages loaded, 4545 targets configured)
    currently loading: @llvm-project//mlir ... (34 packages)
ERROR: /DEFAULT.WORKSPACE.SUFFIX:519:13: fetching cc_autoconf rule //external:local_config_cc: Traceback (most recent call last):
        File "/private/tmp/nix-build-tensorflow-2.10.0-rc0-deps.tar.gz.drv-0/output/external/bazel_tools/tools/cpp/cc_configure.bzl", line 127, column 33, in cc_autoconf_impl
                configure_unix_toolchain(repository_ctx, cpu_value, overriden_tools)
        File "/private/tmp/nix-build-tensorflow-2.10.0-rc0-deps.tar.gz.drv-0/output/external/bazel_tools/tools/cpp/unix_cc_configure.bzl", line 375, column 46, in configure_unix_toolchain
                overriden_tools["ar"] = _find_generic(repository_ctx, "libtool", "LIBTOOL", overriden_tools)
        File "/private/tmp/nix-build-tensorflow-2.10.0-rc0-deps.tar.gz.drv-0/output/external/bazel_tools/tools/cpp/unix_cc_configure.bzl", line 306, column 32, in _find_generic
                auto_configure_fail(msg)
        File "/private/tmp/nix-build-tensorflow-2.10.0-rc0-deps.tar.gz.drv-0/output/external/bazel_tools/tools/cpp/lib_cc_configure.bzl", line 112, column 9, in auto_configure_fail
                fail("\n%sAuto-Configuration Error:%s %s\n" % (red, no_color, msg))
Error in fail:
Auto-Configuration Error: Cannot find libtool or LIBTOOL; either correct your path or set the LIBTOOL environment variable
Analyzing: 2 targets (256 packages loaded, 4545 targets configured)
```

File: pkgs/development/python-modules/tensorflow/default.nix [hunk 2]
```diff
@@ -369,13 +369,13 @@ let
...
         else
-          "sha256-zH3xNFEU2JR0Ww8bpD4mCiorGtao0WVPP4vklVMgS4A=";
+          "sha256-9IZPHo0Rv9TzgkCnIkkk/b6A9FftgXo3D4KjcwrWxuk=";
```
Comment (dmivankov) [inline pkgs/development/python-modules/tensorflow/default.nix:378]:
```suggestion
          "sha256-/oP3+S1p2njHrl/PqIq2NfYm+5cn+jOeLHTTMf64pcA=";
```

**Comment by dmivankov** [comment]
@ofborg build python3Packages.tensorflow

File: pkgs/development/python-modules/tensorflow/default.nix [hunk 3]
```diff
@@ -369,13 +369,13 @@ let
...
-        "sha256-KtVReqHL3zxE8TPrqIerSOt59Mgke/ftoFZKMzgX/u8="
+        "sha256-KtVReqHL3zxE8TPrqIerSOt59Mgke/ftoFZKMzgX/u1="
       else
         if stdenv.isDarwin then
           # FIXME: this checksum is currently wrong, since the tensorflow dependency fetch is broken on darwin
-          "sha256-j2k9Q+k41nq5nP1VjjkkNjXRov1uAda4RCMDMAthjr0="
+          "sha256-j2k9Q+k41nq5nP1VjjkkNjXRov1uAda4RCMDMAthjr1="
         else
-          "sha256-zH3xNFEU2JR0Ww8bpD4mCiorGtao0WVPP4vklVMgS4A=";
+          "sha256-/oP3+S1p2njHrl/PqIq2NfYm+5cn+jOeLHTTMf64pcA=";
```
Comment (dmivankov) [inline pkgs/development/python-modules/tensorflow/default.nix:378]:
```suggestion
          "sha256-9IZPHo0Rv9TzgkCnIkkk/b6A9FftgXo3D4KjcwrWxuk=";
```

**Comment by dmivankov** [comment]
https://github.com/ofborg build python3Packages.tensorflow

**Comment by dmivankov** [comment]
also such logs for darwin
```
do not know how to unpack source archive /nix/store/i8w26qrbdv16w5amyfk369xrrnw3xnj1-source
/nix/store/ffjdm44bpg5h9ydfapsviygp8gylvfh8-stdenv-linux/setup: line 101: /dev/null: No such file or directory
/nix/store/ffjdm44bpg5h9ydfapsviygp8gylvfh8-stdenv-linux/setup: line 82: /dev/null: No such file or directory
/nix/store/ffjdm44bpg5h9ydfapsviygp8gylvfh8-stdenv-linux/setup: line 84: /dev/null: No such file or directory
error: builder for '/nix/store/8iwn5c0jrnk99x5jiasgkw834jix1wgd-tensorflow-2.10.0-rc0-deps.tar.gz.drv' failed with exit code 1;
       last 10 log lines:
       > cp: setting permissions for 'source/tensorflow/python/util': Operation not supported
       > cp: setting permissions for 'source/tensorflow/security': Operation not supported
       > cp: setting permissions for 'source/tensorflow/stream_executor': Operation not supported
       > cp: setting permissions for 'source/tensorflow/tools': Operation not supported
       > cp: setting permissions for 'source/third_party': Operation not supported
       > cp: setting permissions for 'source/tools': Operation not supported
       > do not know how to unpack source archive /nix/store/i8w26qrbdv16w5amyfk369xrrnw3xnj1-source
       > /nix/store/ffjdm44bpg5h9ydfapsviygp8gylvfh8-stdenv-linux/setup: line 101: /dev/null: No such file or directory
       > /nix/store/ffjdm44bpg5h9ydfapsviygp8gylvfh8-stdenv-linux/setup: line 82: /dev/null: No such file or directory
       > /nix/store/ffjdm44bpg5h9ydfapsviygp8gylvfh8-stdenv-linux/setup: line 84: /dev/null: No such file or directory
       For full logs, run 'nix log /nix/store/8iwn5c0jrnk99x5jiasgkw834jix1wgd-tensorflow-2.10.0-rc0-deps.tar.gz.drv'.
```

**Comment by divanorama** [comment]
bazel_5 builds ok on aarch64-darwin

File: pkgs/development/python-modules/tensorflow/default.nix (hunk 1 above)
Comment (dmivankov) [reply pkgs/development/python-modules/tensorflow/default.nix:376]: will revert those as can't update due to unrelated build errors

Comment (dmivankov) [inline pkgs/development/python-modules/tensorflow/default.nix:376]:
```suggestion
          "sha256-j2k9Q+k41nq5nP1VjjkkNjXRov1uAda4RCMDMAthjr0="
```

File: pkgs/development/python-modules/tensorflow/default.nix [hunk 4]
```diff
@@ -369,13 +369,13 @@ let
     fetchAttrs = {
       # cudaSupport causes fetch of ncclArchive, resulting in different hashes
       sha256 = if cudaSupport then
-        "sha256-KtVReqHL3zxE8TPrqIerSOt59Mgke/ftoFZKMzgX/u8="
+        "sha256-KtVReqHL3zxE8TPrqIerSOt59Mgke/ftoFZKMzgX/u1="
```
Comment (dmivankov) [inline pkgs/development/python-modules/tensorflow/default.nix:372]:
```suggestion
        "sha256-KtVReqHL3zxE8TPrqIerSOt59Mgke/ftoFZKMzgX/u8="
```

**Comment by divanorama** [comment]
aarch64-darwin fails for bazel.passthru.tests (bazel 3.7.2)
```
src/main/tools/process-wrapper-legacy.cc:80: "execvp(external/remote_java_tools_darwin/java_tools/ijar/ijar, ...)": Bad CPU type in executable
```
which sounds like https://github.com/bazelbuild/bazel/issues/13944
bazel_5.passthru.tests builds ok.

**Comment by dmivankov** [comment]
Yes, I think it should be good enough now

**Comment by dmivankov** [comment]
closing in favor of 5.3.2 PR https://github.com/NixOS/nixpkgs/pull/198016

**Pull request closed without merging**